server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -519,7 +524,17 @@ public Boolean isLockAcquired(long primaryTerm, long generation) throws IOExcept
 
     // Visible for testing
     Boolean isLockAcquired(String metadataFile) throws IOException {
-        return mdLockManager.isAcquired(FileLockInfo.getLockInfoBuilder().withFileToLock(metadataFile).build());
+        if (lockCache.get(metadataFile) != null) {
+            return true;
+        }
+        boolean lockAcquired = mdLockManager.isAcquired(FileLockInfo.getLockInfoBuilder().withFileToLock(metadataFile).build());
+        if (lockAcquired) {
+            // We do not want to cache info of unlocked metadata files as we do not want to miss a lock getting acquired
+            // on a metadata file between 2 invocations of this method. Caching unlocked files can lead to data
+            // consistency issues.
+            lockCache.put(metadataFile, true);
+        }
```

Comment:
Assuming we want the lock cache to be pessimistic, as in even if the underlying lock is released we want the system to considered it locked for atleast an hour(cache interval)

Reply:
yes